subprojects/core/src/main/java/org/gradle/api/internal/ExtensibleDynamicObject.java

```diff
@@ -48,25 +51,22 @@
     }
 
     private final AbstractDynamicObject dynamicDelegate;
+    private final DefaultExtensionContainer extensions;
+    private final DynamicObject extraPropertiesDynamicObject;
+
     private DynamicObject parent;
-    private Convention convention;
     private DynamicObject beforeConvention;
```

Comment:
Btw. most of these fields and methods are only ever used by `Project`, so another good step would be to split the whole implementation into "normal extensible objects" and "project extensible object".